### PR TITLE
Support turbo mode for better latency but burn more power on CPU & NPU

### PR DESCRIPTION
**Pull request by mamin506**
1. Mailbox channel support three types: MGMT, USER_NORMAL and USER_POLL
- MGMT channel is interrupt based and good for not critical path.
- USER_NORMAL channel is interrupt based and good for critical path.
- USER_POLL channel is polling based and good for critical path which needs the best latency.
2. Support "xrt-smi configure --pmode turbo"
- Only when no context is opened, change pmode from/to turbo are allowed.
- In turbo mode, all contexts are creating USER_POLL mailbox channel for best latency.  
3. Support hwq::poll_command() in shim
4. Add shim_test for polling

|  test name  |  ID (wait in ioctl)   |   ID (polling)  |
|---------|------------------|-----------|
| latency           | 25 | 41 |
| throughput    | 40 | 42 |
| runlist latency | 39 | 43 |
| runlist throughput | 30 | 44 |

### CODE REVIEW

File: src/driver/amdxdna/aie2_pm.c [hunk 1]
```diff
@@ -41,7 +67,7 @@ int aie2_pm_set_mode(struct amdxdna_dev_hdl *ndev, enum amdxdna_power_mode_type
 	if (ndev->pw_mode == target)
 		return 0;
 
-	if (target == POWER_MODE_LOW || target == POWER_MODE_MEDIUM)
+	if (target != POWER_MODE_DEFAULT && target != POWER_MODE_TURBO)
```
Comment (maxzhen) [inline src/driver/amdxdna/aie2_pm.c]: We need to support performance/default/turbo.

File: test/shim_test/io_test.cpp [hunk 2]
```diff
@@ -83,6 +84,22 @@ io_test_init_runlist_cmd(bo* cmd_bo, std::vector<bo*>& cmd_bos)
 }
 
 #define IO_TEST_TIMEOUT 5000 /* millisecond */
+
+void io_test_cmd_wait(hwqueue_handle *hwq, std::shared_ptr<bo> bo)
+{
+    if (io_test_parameters.wait == IO_TEST_POLL_WAIT) {
+        auto start = clk::now();
+        while(!hwq->poll_command(bo->get())) {
+            auto now = clk::now();
```
Comment (maxzhen) [inline test/shim_test/io_test.cpp]: It might be too heavy to call clk::now() for every poll.

**Review by maxzhen — CHANGES_REQUESTED** [review]

File: src/driver/amdxdna/aie2_pm.c (hunk 1 above)
Comment (mamin506) [reply src/driver/amdxdna/aie2_pm.c]: fixed

File: test/shim_test/io_test.cpp (hunk 2 above)
Comment (mamin506) [reply test/shim_test/io_test.cpp]: fixed

**Review by maxzhen — APPROVED** [review]

**Pull request merged**